src/proxy/http/fallback.rs

```diff
@@ -0,0 +1,336 @@
+use bytes::Buf;
+use futures::{Async, Future, Poll};
+use http;
+use hyper::body::Payload;
+use proxy;
+use svc;
+
+use std::{fmt, marker::PhantomData, mem};
+
+#[derive(Debug)]
+pub struct Error<A> {
+    error: proxy::Error,
+    fallback: Option<http::Request<A>>,
+}
+
+pub type Layer<P, F, A> = MakeFallback<svc::Builder<P>, svc::Builder<F>, A>;
```

Comment:
that we do this for Layer and not the others is.... unclear? I'd just redefine it? but i don't care deeply

Reply:
I was initially going to reuse one struct for more of the types but it didn't work for the two Service impls due to coherence. Can replace this with a separate struct.

Reply:
(i just wanted to save on the `impl Clone` boilerplate honestly)